src/languageServerInstaller.ts

```diff
@@ -32,31 +36,45 @@ export class LanguageServerInstaller {
 			isInstalled = false;
 		}
 
-		const currentRelease = await getRelease("terraform-ls", "latest", userAgent);
-		if (isInstalled) {
-			this.reporter.sendTelemetryEvent('foundLsInstalled', { terraformLsVersion: installedVersion });
-			if (semver.gt(currentRelease.version, installedVersion, { includePrerelease: true })) {
-				const selected = await vscode.window.showInformationMessage(`A new language server release is available: ${currentRelease.version}. Install now?`, 'Install', 'Cancel');
-				if (selected !== 'Install') { // selected is undefined if the user closes the message
-					return;
-				}
-			} else {
-				return;
-			}
+		this.release = await this.latestRelease();
+		if (!isInstalled) {
+			return true;
 		}
 
+		this.reporter.sendTelemetryEvent('foundLsInstalled', { terraformLsVersion: installedVersion });
+		const upgrade = semver.gt(this.release.version, installedVersion, { includePrerelease: true });
+		if (upgrade) {
+			const selected = await vscode.window.showInformationMessage(`A new language server release is available: ${this.release.version}. Install now?`, 'Install', 'Cancel');
+			return (selected === "Install");
+		} else {
+			return false; // no upgrade available
+		}
+	}
+
+	public async install(): Promise<void> {
+		this.reporter.sendTelemetryEvent('installingLs', { terraformLsVersion: this.release.version });
 		try {
-			this.reporter.sendTelemetryEvent('installingLs', { terraformLsVersion: currentRelease.version });
-			await this.installPkg(currentRelease, directory, userAgent);
+			await this.installPkg(this.release);
 		} catch (err) {
 			vscode.window.showErrorMessage(`Unable to install terraform-ls: ${err.message}`);
 			throw err;
 		}
 
-		this.showChangelog(currentRelease.version);
+		this.showChangelog(this.release.version);
+	}
+
+	_latestReleasePromise: Promise<Release>;
```

Comment:
I'm not sure we want to cache this result? It seemed like updates were checked for on a schedule now, so I think we don't want to cache the response, that way when a new release goes out, the logic will show a mismatch between what the user has installed and the new latest

Reply:
I'll remove it, I was thinking I would need it between checking whether there's an update and installing it, but I went in a different direction for that part.